lib/actions/create-membership.ts

```diff
@@ -0,0 +1,45 @@
+'use server'
+
+import {Role} from '@prisma/client'
+import prisma from '~/prisma'
+import {getCurrentUser} from '~/utils/session'
+
+// Create membership to join a team by accepting an invitation
+export default async function createMembership(
+	inviteId: string,
+	teamId: string,
```

Comment:
Potential vulnerability: passing the `teamId` here could allow someone to join any team

Luckily (/ good work above), the `teamId` is stored on the `Invite`. Could first do a lookup by `inviteId`.

As well, `userId` could allow someone to **add** any user to their team, or add themselves to another team by hi-jacking the invite ID. Here, we could use `user.id` from the session.

Reply:
Great catch, looking up by `inviteId` is a lot better than passing params.